packages/router/src/lib/routes.ts

```diff
@@ -0,0 +1,129 @@
+/// <reference types="vite/client" />
+
+import type { Type } from '@angular/core';
+import type { Route } from '@angular/router';
+import { defineRouteMeta } from './define-route';
+
+type Module = {
+  default: Type<unknown>;
+  routeMeta?: ReturnType<typeof defineRouteMeta>;
+};
+
+const FILES = import.meta.glob<Module>(['/app/routes/**/*.ts']);
+
+const ROUTES = Object.keys(FILES).sort((a, b) => a.length - b.length);
+const routeConfigs = ROUTES.reduce<Route[]>((routes: Route[], key: string) => {
+  const module = FILES[key];
+
+  const segments = key
+    .replace(/\/app\/routes|\.(js|ts)$/g, '')
+    .replace(/\[\.{3}.+\]/, '**')
+    .replace(/\[([^\]]+)\]/g, ':$1')
+    .split('/')
+    .filter(Boolean);
+
+  segments.reduce((parent, segment, index) => {
```

Comment:
question: this `reduce` return value is not assigned to anything?

Reply:
No, I believe this is just to be able to update the parent route if child routes are found while going through each segment. I'm open to suggestions to clean this up if needed.

Reply:
You're making me pull this PR down, aren't you? 😅 I'll give it a closer look.

Reply:
😎

Reply:
If you could put in some test cases for `routes.ts`, I might be able to trim it down a bit.

Reply:
I added some tests for the routes and pushed it up